### PR TITLE
Feature/sc 4041 studentlist visibility

### PR DESCRIPTION
**Pull request by j-jmnz**
# Description

## Links to Tickets or other pull requests
https://ticketsystem.schul-cloud.org/browse/SC-4041
https://github.com/schul-cloud/schulcloud-server/pull/1346

## Approval for review
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definition of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: helpers/handlebars/middleware.js [hunk 1]
```diff
@@ -227,7 +248,7 @@ module.exports = (req, res, next) => {
 				$limit: 10,
 				$sort: '-createdAt',
 			},
-		}).catch(_ => []);
+		}).catch((_) => []);
```
Comment (CeEv) [inline helpers/handlebars/middleware.js:251]: _ can removed

File: helpers/authentication.js [hunk 2]
```diff
@@ -97,11 +97,15 @@ const populateCurrentUser = (req, res) => {
...
-			},
-		}).then((data) => {
+		return Promise.all([
+			api(req).get(`/users/${payload.userId}`),
+			api(req).get(`/roles/user/${payload.userId}`),
+		]).then(([user, roles]) => {
+			const data = {
+				...user,
+				roles,
+				permissions: [...new Set(roles.map(({ permissions }) => permissions))][0],
```
Comment (CeEv) [inline helpers/authentication.js:107]: Please check [0] it look wrong.

File: controllers/administration.js [hunk 3]
```diff
@@ -2975,7 +3034,7 @@ router.use(
 	'/school',
 	permissionsHelper.permissionsChecker(['ADMIN_VIEW', 'TEACHER_CREATE'], 'or'),
 	async (req, res) => {
-		const [school, totalStorage, schoolMaintanance] = await Promise.all([
+		const [school, totalStorage, schoolMaintanance, studentVisibility] = await Promise.all([
```
Comment (CeEv) [inline controllers/administration.js:3037]: /school/teacher/permissions return an object that includes isEnabled as key?

File: controllers/administration.js [hunk 4]
```diff
@@ -870,7 +871,7 @@ const getConsentStatusIcon = (consentStatus, isTeacher = false) => {
 // teacher admin permissions
 router.get(
 	'/',
-	permissionsHelper.permissionsChecker(['ADMIN_VIEW', 'STUDENT_LIST'], 'or'),
+	permissionsHelper.permissionsChecker(['ADMIN_VIEW', 'STUDENT_LIST', 'TEACHER_LIST'], 'or'),
```
Comment (CeEv) [inline controllers/administration.js:874]: If the permission TEACHER_LIST exist, then the admin area can be entered?

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5660832&bid=18530109)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
- Added 1
           

Complexity increasing per file
==============================
- helpers/authentication.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5660832&bid=18530109)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
- Added 1
           

Complexity increasing per file
==============================
- helpers/authentication.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5660832&bid=18530109)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
- Added 1
           

Complexity increasing per file
==============================
- helpers/authentication.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5660832&bid=18530109)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
- Added 1
           

Complexity increasing per file
==============================
- helpers/authentication.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5660832&bid=18530109)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
- Added 1
           

Complexity increasing per file
==============================
- helpers/authentication.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5660832&bid=18530109)

**Pull request merged**